### ISSUE TITLE
Add the ability to show logged-in users in Specify 7

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#3608]
> This feature is available in S6, from the top menu option 'System' > 'Schema Configuration':
> A popup menu appears, titled 'Other users are logged on' 
> 
> For context, we are trying to troubleshoot a long-running query. In order to better understand how this operation is starting, it would be helpful to know which users are logged onto the system at the time.

**Requested by:** Garth at the University of Michigan

_There must be an issue already for this, so please close it as a duplicate if/when you find the original._

**Comment by maxxxxxdlp** [comment]
I don't think we have an issue for this. I also don't think we can do this without a schema change.

Also, unlike in sp6, where there is only one instance open at one point, and it explicitly tells the database when it disconnects, there could be multiple browser tabs open. There is a way to execute code (like a network request to the back-end API) the moment before the tab is closed, but that is not working reliably in all cases (i.e, imagine closing the browser on ios, or loosing network connection) and would require additional logic for when there are multiple tabs, or multiple browsers or multiple devices logged in to same user.

The simplest solution would be to have back-end update some sort of "Last online" column any time front-end requests notifications (since that request is sent regularly, and only while the tab is open). Then, we could say if "Last online" column value is within the last 5 minutes, then consider the user to be online

**Comment by grantfitzsimmons** [comment]
>The simplest solution would be to have back-end update some sort of "Last online" column any time front-end requests notifications (since that request is sent regularly, and only while the tab is open). Then, we could say if "Last online" column value is within the last 5 minutes, then consider the user to be online

I think that's a wonderful idea.

**Comment by grantfitzsimmons** [comment]
>Thanks for looking into this one for us. This would be a useful feature to have for troubleshooting. 
Otherwise, we're left polling the users and searching the audit logs when things go wrong and trying to figure out what happened.

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/online-user-status-in-specify-7/2093/2

**Comment by grantfitzsimmons** [comment]
From [Philippe Verley at IRD](https://discourse.specifysoftware.org/t/online-user-status-in-specify-7/2093?u=grant):

> I still have a limited understanding of the Specify-7 server architecture :baby:
> 
> :question: would it be possible to show in [Security and Accounts](https://discourse.specifysoftware.org/t/security-and-accounts/539) page, next to the users, who is:
> 
> :green_circle: logged-in and active
> :yellow_circle: logged-in but inactive
> :black_circle: logged-off
> Related to the [Maintenance mode request](https://discourse.specifysoftware.org/t/maintenance-mode-in-specify-7/2092), I would find it useful as an Institution admin to know at a glance the activity on the system (for instance whether it is a good time to launch an update, a backup, a clean-up, etc.)

+1 from Heryk as well

**Comment by grantfitzsimmons** [comment]
Requested again by the College of Idaho

**Comment by bronwyncombs** [comment]
Requested by Marion @ Museum für Naturkunde Berlin via [BoldDesk](https://specifysoftware.bolddesk.com/support/tickets/1033): 

> As I explained, it’s very difficult for me to know who’s doing what. Even if I send messages to find out who’s online when slowdowns occur, I never get a reply. Similarly, even if I explain what might be causing the slowdowns, I’m not sure everyone has that information. That is why I would like to know if it is possible to identify who is doing what in the database, so that I can inform the relevant people in real time, thereby enabling all users to work under optimal conditions.
